pkg/secretstorage/vaultstorage/vaultcli/cli.go

```diff
@@ -18,40 +18,64 @@ import (
 	VaultDataPathPrefix            string                       `arg:"--vault-data-path-prefix, env" default:"spi" help:"Path prefix in Vault token storage under which all SPI data will be stored. No leading or trailing '/' should be used, it will be trimmed."`
 }
 
-// VaultStorageConfigFromCliArgs returns an instance of the VaultStorageConfig with some fields initialized from
-// the corresponding CLI arguments. Notably, the VaultStorageConfig.MetricsRegisterer is NOT configured, because this
-// cannot be done using just the CLI arguments.
-func VaultStorageConfigFromCliArgs(args *VaultCliArgs) *vaultstorage.VaultStorageConfig {
-	return &vaultstorage.VaultStorageConfig{
+//// VaultStorageConfigFromCliArgs returns an instance of the VaultStorageConfig with some fields initialized from
+//// the corresponding CLI arguments. Notably, the VaultStorageConfig.MetricsRegisterer is NOT configured, because this
+//// cannot be done using just the CLI arguments.
+//// func VaultStorageConfigFromCliArgs(args *VaultCliArgs) *vaultstorage.VaultStorageConfig {
+//	return &vaultstorage.VaultStorageConfig{
+//		Host:                        args.VaultHost,
+//		AuthType:                    args.VaultAuthMethod,
+//		Insecure:                    args.VaultInsecureTLS,
+//		Role:                        args.VaultKubernetesRole,
+//		ServiceAccountTokenFilePath: args.VaultKubernetesSATokenFilePath,
+//		RoleIdFilePath:              args.VaultApproleRoleIdFilePath,
+//		SecretIdFilePath:            args.VaultApproleSecretIdFilePath,
+//		DataPathPrefix:              strings.Trim(args.VaultDataPathPrefix, "/"),
+//	}
+//}
+
+func CreateVaultStorage(ctx context.Context, reader client.Reader, args *VaultCliArgs) (secretstorage.SecretStorage, error) {
+	vaultConfig := &vaultstorage.VaultStorageConfig{
 		Host:                        args.VaultHost,
 		AuthType:                    args.VaultAuthMethod,
 		Insecure:                    args.VaultInsecureTLS,
 		Role:                        args.VaultKubernetesRole,
 		ServiceAccountTokenFilePath: args.VaultKubernetesSATokenFilePath,
-		RoleIdFilePath:              args.VaultApproleRoleIdFilePath,
-		SecretIdFilePath:            args.VaultApproleSecretIdFilePath,
 		DataPathPrefix:              strings.Trim(args.VaultDataPathPrefix, "/"),
 	}
-}
 
-func CreateVaultStorage(ctx context.Context, args *VaultCliArgs) (secretstorage.SecretStorage, error) {
-	vaultConfig := VaultStorageConfigFromCliArgs(args)
+	if args.VaultAuthMethod == vaultstorage.VaultAuthMethodApprole {
+
+		secret := &corev1.Secret{}
+		key := client.ObjectKey{Namespace: "remotesecret", Name: args.VaultAppRoleSecretName}
+		err := reader.Get(ctx, key, secret)
+		if err != nil {
+			return &vaultstorage.VaultSecretStorage{
+				Config: vaultConfig,
+			}, errors.Wrap(err, "failed to read secret")
```

Comment:
could you pls explain why `fmt.Errorf()` is not enough here  and we're forced to add the new dependency?

Reply:
Changed it to fmt.Errorf(), thx.